i6_models/config.py

```diff
@@ -51,3 +53,22 @@ def _validate_types(self) -> None:
 
     def __post_init__(self) -> None:
         self._validate_types()
+
+
+ConfigType = TypeVar("ConfigType", bound=ModelConfiguration)
+ModuleType = TypeVar("ModuleType", bound=nn.Module)
+
+
+@dataclass
+class SubassemblyFactory(Generic[ConfigType, ModuleType]):
```

Comment:
I don't like the name so much. This is not really just for subassemblies. This can (and probably will) be used for any of our modules which have this generic interface where `__init__` accepts the config object.
```suggestion
class ModuleFactoryV1(Generic[ConfigType, ModuleType]):
```
?